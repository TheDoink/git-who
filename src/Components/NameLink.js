import React, { Component } from 'react';

class NameLink extends Component {

  handleNameClick() {
    // Set a storage object for later remembering
    sessionStorage.setItem('gitUser', this.props.userName);

    window.location = "/";
  }  

  render() {
    
    return (
      <span>
        <a href="#" onClick={this.handleNameClick.bind(this)}>
          {this.props.userName}
        </a>
      </span>
    );
  }
}

export default NameLink;
