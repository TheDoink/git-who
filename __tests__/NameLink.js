// var NameLink = require('../src/Components/NameLink');
import React from 'react';
import renderer from 'react-test-renderer';
import NameLink from '../src/Components/NameLink';

it('lines up with the snapshot', () => {
  let myLink = renderer.create(<NameLink userName="steve"></NameLink>).toJSON();
  expect(myLink).toMatchSnapshot();
});
