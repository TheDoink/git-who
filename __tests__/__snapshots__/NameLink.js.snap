// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lines up with the snapshot 1`] = `
<span>
  <a
    href="#"
    onClick={[Function]}
  >
    steve
  </a>
</span>
`;
